### PR TITLE
Adjust docs to sidecar injection annotation deprecation

### PR DESCRIPTION
**Pull request by nataliasitko**
Changes proposed in this pull request:

- Change sidecar injection annotation to sidecar injection label in the Kyma docs

**Related issue(s)**
- https://github.com/kyma-project/kyma/issues/16374

### CODE REVIEW

File: docs/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ You can enable sidecar proxy injection for either an entire Namespace or a singl
    kubectl label namespace {YOUR_NAMESPACE} istio-injection=enabled
    ```
 
-* To enable sidecar proxy injection for a Deployment, add this to the Deployment configuration file as either a label or an annotation: `sidecar.istio.io/inject: "true"`
+* To enable sidecar proxy injection for a Deployment, add this label to the Deployment configuration file: `sidecar.istio.io/inject: "true"`
 
-Note that the Namespace label takes precedence over the Pod label or annotation.
+Note that the Namespace label takes precedence over the Pod label.
```
Comment (barchw) [inline docs/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md:17]:
```suggestion
Note that if any of Namespace or Pod label disables Istio injection, the sidecar proxy will not be injected.
```

Comment (barchw) [reply docs/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md:17]: That should be the case if I'm looking correctly
| default policy (enableNamespacesByDefault) | namespaceSelector (label: "istio-injection"]) | Pod label(sidecar.istio.io/inject) | Sidecar injected? |
|:------------------------------------------:|:---------------------------------------------:|:-------------------------------------------------:|:-----------------:|
|            true	            		             |                 enabled						                 |                    true 						                    |      yes				      |
|                  true			                   |                 enabled						                 |                    false						                    |      no				       |
|                true     		                 |                disabled						                 |                    true						                     |      no				       |
|                  true			                   |                disabled						                 |                    false						                    |      no				       |
|                  true			                   |                  not present                  |                    true						                     |      yes				      |
|                  true			                   |                not present				                |                    false						                    |      no				       |
|                  true			                   |                not present				                |                  not present 			                  |      yes				      |
|                 false   		                 |                 enabled						                 |                    true 						                    |      yes				      |
|                 false   		                 |                 enabled						                 |                    false						                    |      no				       |
|                 false   		                 |                disabled						                 |                    true						                     |      no				       | 
|                 false   		                 |                disabled						                 |                    false						                    |      no				       |
|                 false   		                 |                not present				                |                    true						                     |      yes				      |
|                 false   		                 |                not present				                |                    false						                    |      no				       |
|                 false   		                 |                not present				                |                  not present			                   |      no				       |

Comment (nataliasitko) [reply docs/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md:17]: Could you provide the link to the docs containing that table? I think we could link it as a further reference

**Review by barchw — APPROVED** [review]

File: docs/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md [hunk 2]
```diff
@@ -12,9 +12,9 @@ You can enable sidecar proxy injection for either an entire Namespace or a singl
    kubectl label namespace {YOUR_NAMESPACE} istio-injection=enabled
    ```
 
-* To enable sidecar proxy injection for a Deployment, add this to the Deployment configuration file as either a label or an annotation: `sidecar.istio.io/inject: "true"`
+* To enable sidecar proxy injection for a Deployment, add this label to the Deployment configuration file: `sidecar.istio.io/inject: "true"`
 
-Note that the Namespace label takes precedence over the Pod label or annotation.
+Note that if the sidecar proxy injection is disabled at the Namespace level or the `sidecar.istio.io/inject` label on a Pod is set to false, the sidecar proxy is not injected.
```
Comment (grego952) [inline docs/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md:17]:
```suggestion
Note that if the sidecar proxy injection is disabled at the Namespace level or the `sidecar.istio.io/inject` label on a Pod is set to `false`, the sidecar proxy is not injected.
```

**Review by grego952 — APPROVED** [review]

**Pull request merged**